python_pellet/agent.py

```diff
@@ -124,72 +154,115 @@ def eelearn(self, ee_memory, batch_size=1000):
+				aux_target_back = self.targetEEnet(merged_states_back)
 
-			for i, s2 in enumerate(partition_memory):
-				max_distance = np.NINF
-				state_to_ref = []
-				s2_to_ref = []
-				ref_to_state = []
-				ref_to_s2 = []
-				for refrence in partition_memory[:5]:
-					state_to_ref.append(merge_states_for_comparason(state, refrence[0]))
-					s2_to_ref.append(merge_states_for_comparason(s2[0], refrence[0]))
-					ref_to_state.append(merge_states_for_comparason(refrence[0], state))
-					ref_to_s2.append(merge_states_for_comparason(refrence[0], s2[0]))
-
-				state_to_ref = torch.cat(state_to_ref)
-				s2_to_ref = torch.cat(s2_to_ref)
-				#TODO HERE
-				forward = torch.sum(self.EEnet(state_to_ref) - self.EEnet(s2_to_ref), dim=1)
-
-				ref_to_state = torch.cat(ref_to_state)
-				ref_to_s2 = torch.cat(ref_to_s2)
-				#TODO HERE
-				backward = torch.sum(self.EEnet(ref_to_state) - self.EEnet(ref_to_s2), dim=1)
-
-				max_distance = torch.max(torch.max(torch.stack([forward, backward], dim=1), 1)[0], 0)[0].item()
-
-				if max_distance < min_distance:
-					min_distance = max_distance
-					index = i
+				aux_forward = torch.mean(self.mse(aux_forward, aux_target_forward), dim=1)
+				aux_back = torch.mean(self.mse(aux_back, aux_target_back), dim=1)
+
+				novelties = torch.max(torch.stack([aux_forward, aux_back], dim=1), 1)[0]
+
+				argmin_value, argmin_index = torch.min(novelties, 0)
+				current_partition = partition_memory[argmin_index.item()]
+				min_distance = argmin_value.item()
+			else:
+				for i, s2 in enumerate(partition_memory):
+					max_distance = np.NINF
+					state_to_ref = []
+					s2_to_ref = []
+					ref_to_state = []
+					ref_to_s2 = []
+					for refrence in partition_memory[:5]:
+						state_to_ref.append(merge_states_for_comparason(state, refrence[0]))
+						s2_to_ref.append(merge_states_for_comparason(s2[0], refrence[0]))
+						ref_to_state.append(merge_states_for_comparason(refrence[0], state))
+						ref_to_s2.append(merge_states_for_comparason(refrence[0], s2[0]))
+
+					state_to_ref = torch.cat(state_to_ref)
+					s2_to_ref = torch.cat(s2_to_ref)
+					forward = torch.sum(self.EEnet(state_to_ref) - self.EEnet(s2_to_ref), dim=1)
+
+					ref_to_state = torch.cat(ref_to_state)
+					ref_to_s2 = torch.cat(ref_to_s2)
+					backward = torch.sum(self.EEnet(ref_to_state) - self.EEnet(ref_to_s2), dim=1)
+
+					max_distance = torch.max(torch.max(torch.stack([forward, backward], dim=1), 1)[0], 0)[0].item()
+
+					if max_distance < min_distance:
+						min_distance = max_distance
+						index = i
 
 			visited_prime = copy.deepcopy(visited)
 
 			if visited[0][index] is None:
-				visited[0][index] = torch.tensor([partition_memory.calc_pellet_reward(partition_memory[index][1])], device=self.device)
+				visited[0][index] = torch.tensor([partition_memory.calc_pellet_reward(partition_memory[index][1])],
+												device=self.device)
 
 			return visited, visited_prime, min_distance
```

Comment:
method find_current_partition()
Same as others, actual differences are:

* added should_use_rnd check, which changes calc in qlearn